### PR TITLE
azure check access

### PR DESCRIPTION
**Pull request by faiq**

### CODE REVIEW

File: azure.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/arm/containerregistry"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+type Azure struct {
```
Comment (termie) [inline azure.go]: are all of these required?

File: azure.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	tenantID          string
+	resourceGroupName string
+	registryName      string
+	registryURL       string
+	regClient         containerregistry.RegistriesClient
+	dockerUsername    string
+	dockerPassword    string
+}
+
+func NewAzure(clientID, clientSecret, subscriptionID, tenantID, resourceGroupName, registryName, registryURL string) (*Azure, error) {
```
Comment (termie) [inline azure.go]: we're edging towards the realm of "AzureOptions" struct, up to you

File: azure.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		username = a.dockerUsername
+	}
+	return username
+}
+
+func (a *Azure) Repository(repository string) string {
+	return fmt.Sprintf("%s/%s", a.registryURL, repository)
+}
+
+func newServicePrincipalTokenFromCredentials(c map[string]string, scope string) (*azure.ServicePrincipalToken, error) {
```
Comment (hatchan) [inline azure.go]: Look like the map can be used as simple arguments, ie:
```
newServicePrincipalTokenFromCredentials(clientID, clientSecret, tenantID, scope)
```

File: azure.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	oauthConfig, err := azure.PublicCloud.OAuthConfigForTenant(c["AZURE_TENANT_ID"])
+	if err != nil {
+		panic(err)
```
Comment (hatchan) [inline azure.go]: Do not panic, return nil and the error

File: azure.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	resourceGroupName string
+	registryName      string
+	registryURL       string
+	regClient         containerregistry.RegistriesClient
+	dockerUsername    string
+	dockerPassword    string
+}
+
+func NewAzure(clientID, clientSecret, subscriptionID, tenantID, resourceGroupName, registryName, registryURL string) (*Azure, error) {
+	c := map[string]string{
```
Comment (hatchan) [inline azure.go]: This map is not used, except for its values so those can be inlined and this map can be removed.

File: azure.go [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/arm/containerregistry"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+type Azure struct {
+	clientID          string
```
Comment (hatchan) [inline azure.go]: Looks like that there are a couple of properties which are only used in the `NewAzure` func (as far as I can tell). So we do not need to define them on the `Azure` struct. Properties that I found: `clientID`, `clientSecret`, `subscriptionID`, `tenantID`.

File: azure.go (hunk 4 above)
Comment (faiq) [inline azure.go]: lol whoops - debugging code got here somehow!

**Pull request merged**